### PR TITLE
Add arch_def generation for CRD

### PR DESCRIPTION
**Pull request by dhower-qc**

### CODE REVIEW

File: lib/arch_def.rb [hunk 1]
```diff
@@ -352,6 +354,21 @@ def implemented_extensions
...
   end
 
+  # @return [Array<ExtensionRequirement>] List of extensions that are explicitly required by an arch def
+  def mandatory_extensions
+    raise "mandatory_extensions is only valid for a partially configured defintion" unless partially_configured?
+
+    return @mandatory_extensions unless @mandatory_extensions.nil?
+
+    @mandatory_extensions = []
+    if @arch_def.key?("mandtory_extensions")
```
Comment (james-ball-qualcomm) [inline lib/arch_def.rb:364]: Typo: mandtory

**Review by james-ball-qualcomm — CHANGES_REQUESTED** [review]
Given the typo I found, I'm wondering how you tested it.

As a second question, is there anything I need to change in the CRD Ruby or Template to accommodate the change?

File: lib/arch_def.rb (hunk 1 above)
Comment (dhower-qc) [reply lib/arch_def.rb:364]: thanks.

Comment (dhower-qc) [reply lib/arch_def.rb:364]: I didn't test it; I see now that it's not obvious at all, but this PR is a "draft"

**Comment by dhower-qc** [comment]
now it has been tested, and the PR is active

**Pull request merged**